internal/pkg/deploy/cloudformation/stack/backend_svc.go

```diff
@@ -212,6 +239,18 @@ func (s *BackendService) httpLoadBalancerTarget() (targetContainer *string, targ
 		targetPort = s.manifest.Sidecars[aws.StringValue(targetContainer)].Port
 	}
 
+	// Route load balancer traffic to the target_port if mentioned.
+	if s.manifest.RoutingRule.TargetPort != nil {
+		targetPort = aws.String(strconv.Itoa(aws.IntValue(s.manifest.RoutingRule.TargetPort)))
+		exposedPorts, err := s.manifest.ExposedPorts()
+		if err == nil {
+			for _, portConfig := range exposedPorts {
+				if portConfig.Port == aws.IntValue(s.manifest.RoutingRule.TargetPort) {
+					targetContainer = aws.String(portConfig.ContainerName)
+				}
+			}
+		}
```

Comment:
I feel like if they only specify target port we should just assume target container is the main container 🤔 And we need manifest validation making sure they don't specify the same port.

Reply:
[Here](https://github.com/aws/copilot-cli/pull/4192#discussion_r1042810508) is some explanation on what are the assumptions we want consider when only `target_port` is mentioned. Do let me know if that solves your concern.

Reply:
What would happen if a user has an incorrectly configured manifest?
```yaml
http:
  path: '/'
  target_port: 80
  target_container: another_sidecar

sidecars:
  nginx:
    port: 80
  another_sidecar:
    port: 8080
```